bpf: revert changes to metrics directions contants

jibi opened the pull request:
This commit partially reverts commit 38ab8f0 ("bpf: add support for
SERVICE metrics") by bringing the `METRIC_INGRESS` and `METRIC_EGRESS`
constants back to their original values.

In addition to that, it introduces the `ct_to_metrics_dir` helper which
allows to convert a CT direction into the corresponding metric one.

This mitigates any upgrade/downgrade impact caused by having keys with 2
different formats on the metrics map.

Fixes: #14175

fristonio commented on bpf/lib/metrics.h:
```diff
@@ -42,4 +42,23 @@ static __always_inline void update_metrics(__u32 bytes, __u8 direction,
+/**
+ * ct_to_metrics_dir
+ * @direction:	1: Ingress 2: Egress 3: Service
+ * Convert a CT direction into the corrisponding one for metrics.
```
```suggestion
 * Convert a CT direction into the corresponding one for metrics.
```

fristonio reviewed (approved):
LGTM :rocket: 
There is a typo in the PR description can you fix that as well. :sweat_smile:

jibi committed:
bpf: revert changes to metrics directions contants

This commit partially reverts commit 38ab8f0f35ba ("bpf: add support for
SERVICE metrics") by bringing the `METRIC_INGRESS` and `METRIC_EGRESS`
constants back to their original values.

In addition to that, it introduces the `ct_to_metrics_dir` helper which
allows to convert a CT direction into the corresponding metric one.

This mitigates any upgrade/downgrade impact caused by having keys with 2
different formats on the metrics map.

Fixes: #14175

Signed-off-by: Gilberto Bertin <gilberto@isovalent.com>
